livecheck/main.py

```diff
@@ -344,8 +339,12 @@ def do_main(*, auto_update: bool, cat: str, ebuild_version: str, parsed_uri: Par
     if tf := settings.transformations.get(cp, None):
         results = [tf(x) for x in results]
     logger.debug(f'Result count: {len(results)}')
-    top_hash = (sorted(results, key=cmp_to_key(special_vercmp), reverse=True)
-                if use_vercmp else results)[0]
+    try:
+        top_hash = (sorted(results, key=cmp_to_key(special_vercmp), reverse=True)
+                    if use_vercmp else results)[0]
+    except IndexError:
```

Comment:
Use the `next()` function to get the first result from an iterator.

```suggestion
    try:
        top_hash = next(sorted(results, key=cmp_to_key(special_vercmp), reverse=True)
                               if use_vercmp else results))
    except StopIteration:
```

Reply:
Show this error
<pre>
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/fran/projectos/livecheck/livecheck/__main__.py", line 3, in <module>
    main()
  File "/usr/lib/python3.12/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fran/projectos/livecheck/livecheck/main.py", line 486, in main
    do_main(r=r,
  File "/home/fran/projectos/livecheck/livecheck/main.py", line 343, in do_main
    top_hash = next((sorted(results, key=cmp_to_key(special_vercmp), reverse=True))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'list' object is not an iterator
</pre>

Reply:
Sorry that suggestion was wrong. Inside next() it needs the conditional expression (ternary) as well.